### ISSUE TITLE
Rework WalletConnect Alephium provider and wallets integration to solve major user problems

### ISSUE THREAD

**Issue by nop33** [alephium/alephium-frontend#928]
Every now and again we get reports from frustrated users that WalletConnect doesn't work: It doesn't respond to events users trigger in the dApp (the mobile and desktop wallet do not receive the session proposal or request events).

I thought we could look into the updated WalletConnect (now ReOwn) docs and maybe use the WalletKit or AppKit. A deeper look into the [Alephium Provider](https://github.com/alephium/alephium-web3/tree/master/packages/walletconnect) might also be necessary.

**Comment by nop33** [comment]
Have we considered using https://docs.reown.com/advanced/providers/universal? Is that something that makes sense?

**Comment by nop33** [comment]
Something else that might be interesting is this: https://docs.reown.com/appkit/javascript/core/custom-networks

**Comment by nop33** [comment]
> TL;DR: Maybe we should update our `@alephium/walletconnect-provider` and `@alephium/web3-react` to use `@reown/...` SDKs.

---

At the new docs homepage of WalletConnect (Reown) we see these 2 SDKs:

https://docs.reown.com/

<img width="899" alt="image" src="https://github.com/user-attachments/assets/1d548dcb-e7ed-45ce-b5c2-831de6f92649">


Clicking on either `React` or `JavaScript` we see some code:

```js
import { createAppKit } from '@reown/appkit'
import { mainnet, arbitrum } from '@reown/appkit/networks'
import { WagmiAdapter } from '@reown/appkit-adapter-wagmi'

// 1. Get a project ID at https://cloud.reown.com
const projectId = 'YOUR_PROJECT_ID'

// 2. Set up Wagmi adapter
const wagmiAdapter = new WagmiAdapter({
  projectId,
  networks
})

// 3. Create the modal
const modal = createAppKit({
  adapters: [wagmiAdapter],
  networks: [mainnet, arbitrum],
  metadata,
  projectId,
  features: {
    analytics: true // Optional - defaults to your Cloud configuration
  }
})

modal.open()
```

Maybe we should replace the old WC packages with this one and provide customized adapters and networks:
- custom networks: https://docs.reown.com/appkit/javascript/core/custom-networks#2-creating-a-custom-chain-object
- `ChainAdapter`: https://github.com/reown-com/appkit/blob/main/packages/core/src/utils/TypeUtil.ts#L918
- `createAppKit`: https://github.com/reown-com/appkit/blob/26e9f12aef1d5e32f7814be189de5d405903b378/packages/appkit/exports/react.ts#L25
- `AppKitOptions`: https://github.com/reown-com/appkit/blob/main/packages/appkit/src/utils/TypesUtil.ts#L5
